src/main/java/com/ericsson/eiffel/remrem/semantics/validator/EiffelValidator.java

```diff
@@ -72,41 +88,133 @@ public EiffelValidator(String schemaResourceName, String eventType, List<String>
+            final String message = "Cannot validate given JSON string";
             log.debug(message, e);
             throw new EiffelValidationException(message, e);
         }
     }
+    private void handleErrorReport(JsonObject jsonObjectInput, ProcessingReport report) throws EiffelValidationException {
+        if (!report.isSuccess()) {
+            log.error("Eiffel message validation failed");
+            log.error(jsonObjectInput.toString());
+            throw new EiffelValidationException(getErrorsList(report));
+        }
+    }
+    /**
+     * removeErrorProperties for removing the validation failures from Eiffel event and list the validation failures
+     * @param report
+     * @param jsonObjectInput
+     * @param remremGenerateFailures
+     * @return
+     * @throws ValidationException
+     */
+    private String removeErrorProperties(ProcessingReport report, JsonObject jsonObjectInput, JsonArray remremGenerateFailures) throws EiffelValidationException {
+        JsonParser parser = new JsonParser();
+        DocumentContext doc = JsonPath.parse(jsonObjectInput.toString());
+        for (ProcessingMessage processingMessage : report) {
+            if (LogLevel.ERROR.equals(processingMessage.getLogLevel())) {
+                JsonElement element = parser.parse(processingMessage.asJson().toString());
+                if (element.getAsJsonObject().get(KEYWORD).getAsString().equals(REQUIRED) || element.getAsJsonObject().get(KEYWORD).getAsString().equals(TYPE)) {
+                    throw new EiffelValidationException(getErrorsList(report));
+                }
+                String errorPath = element.getAsJsonObject().get(INSTANCE).getAsJsonObject().get(POINTER)
+                        .getAsString();
+                JsonObject addValidationFailurs = addValidationFailures(element, processingMessage.getMessage());
+                remremGenerateFailures.add(addValidationFailurs);
+                doc.delete(getPath(errorPath));
+            }
+        }
+        return doc.jsonString();
+    }
+    
+    /**
+     * getPath for create the DocumentContext path from validation failure path
+     * example 
+     * 1) meta/tags/0 --> $.meta.tags[0]
+     * 2) links/0/target --> $.links[0]
+     * 3) /data/outcome/conclusion --> $.data.outcome.conclusion
+     * @param errorPath
+     * @return path
+     */
+    private String getPath(String errorPath) {
+        final Pattern pattern = Pattern.compile(REGEX_PATH, Pattern.MULTILINE);
+        final Matcher matcher = pattern.matcher(errorPath);
+        while (matcher.find()) {
+            if(matcher.group(0).length()>1) {
+                errorPath = errorPath.substring(0, errorPath.indexOf(matcher.group(0))+matcher.group(0).length());
+                errorPath = errorPath.replaceAll(matcher.group(0), "["+matcher.group(0).substring(1)+"]");
+            }
+        }
+        return "$" + errorPath.replace(SLASH, DOT);
+    }
```

Comment:
No tests for this code?

Reply:
Its private method and we don't have the unit tests for private methods. but we added unit tests for validations with lenient validations.